server/src/main/kotlin/suwayomi/tachidesk/opds/impl/OpdsFeedBuilder.kt

```diff
@@ -0,0 +1,1016 @@
+                entries.addAll(
+                    languages.map { entry: OpdsLanguageNavEntry ->
+                        OpdsEntryXml(
+                            id = "urn:suwayomi:navigation:language:${entry.id}",
+                            title = entry.title,
+                            updated = currentFormattedTime(),
+                            link =
+                                listOf(
+                                    OpdsLinkXml(
+                                        OpdsConstants.LINK_REL_SUBSECTION,
+                                        "$baseUrl/language/${entry.id}?lang=${uiLocale.toLanguageTag()}",
+                                        OpdsConstants.TYPE_ATOM_XML_FEED_ACQUISITION,
+                                        entry.title,
+                                    ),
+                                ),
+                        )
+                    },
+                )
+            }
+        return OpdsXmlUtil.serializeFeedToString(builder.build())
+    }
+
+    // --- Specific Acquisition Feed Generators ---
+
+    fun getMangaFeed(
+        mangaId: Int,
+        baseUrl: String,
+        pageNum: Int,
+        sortParam: String?,
+        filterParam: String?,
+        locale: Locale,
+    ): String {
+        val mangaDetails =
+            MangaRepository.getMangaDetails(mangaId)
+                ?: return buildNotFoundFeed(
+                    baseUrl,
+                    "manga/$mangaId",
+                    MR.strings.opds_error_manga_not_found.localized(locale, mangaId.toString()),
+                    locale,
+                )
+
+        val (sortColumn, currentSortOrder) =
+            when (sortParam?.lowercase()) {
+                "asc", "number_asc" -> ChapterTable.sourceOrder to SortOrder.ASC
+                "desc", "number_desc" -> ChapterTable.sourceOrder to SortOrder.DESC
+                "date_asc" -> ChapterTable.date_upload to SortOrder.ASC
+                "date_desc" -> ChapterTable.date_upload to SortOrder.DESC
+                else -> ChapterTable.sourceOrder to (serverConfig.opdsChapterSortOrder.value ?: SortOrder.ASC)
+            }
+        val currentFilter = filterParam?.lowercase() ?: if (serverConfig.opdsShowOnlyUnreadChapters.value) "unread" else "all"
+
+        val (chapterEntries, totalChapters) =
+            ChapterRepository.getChaptersForManga(
+                mangaId,
+                pageNum,
+                sortColumn,
+                currentSortOrder,
+                currentFilter,
+            )
+
+        val actualSortParamForLinks =
+            sortParam ?: run {
+                val prefix = if (sortColumn == ChapterTable.sourceOrder) "number" else "date"
+                val suffix = if (currentSortOrder == SortOrder.ASC) "asc" else "desc"
+                "${prefix}_$suffix"
+            }
+
+        val builder =
+            FeedBuilderInternal(
+                baseUrl = baseUrl,
+                idPath = "manga/$mangaId/chapters",
```

Comment:
idPath seems to point to an invalid endpoint (pagination doesn't work anymore)
```suggestion
                idPath = "manga/$mangaId",
```

Reply:
Those path changes were part of the OPDS API restructuring I forgot to push earlier.